spark/src/main/java/org/apache/zeppelin/spark/SparkInterpreter.java

```diff
@@ -340,22 +340,19 @@ public boolean isYarnMode() {
    * Create SparkSession
    */
   public Object createSparkSession() {
-    logger.info("------ Create new SparkContext {} -------", getProperty("master"));
+    // use local mode for embedded spark mode when spark.master is not found
+    if (!conf.contains("spark.master")) {
+      conf.setMaster("local");
```

Comment:
nit: you can also use http://spark.apache.org/docs/latest/api/scala/index.html#org.apache.spark.SparkConf@setIfMissing(key:String,value:String):org.apache.spark.SparkConf